src/mime.rs

```diff
@@ -1,50 +1,88 @@
+use std::borrow::Cow;
+use thiserror::Error;
+
 /// List of allowed mimes.
-pub static ALLOWED_MIME_TYPES: [&str; 3] =
+pub static ALLOWED_TEXT_MIME_TYPES: [&str; 3] =
     ["text/plain;charset=utf-8", "UTF8_STRING", "text/plain"];
 
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Unsupported mime type")]
+    Unsupported,
+}
+
 /// Mime type supported by clipboard.
-#[derive(Clone, Copy, Eq, PartialEq, Debug)]
+#[derive(Clone, Eq, PartialEq, Debug, Default)]
+#[repr(u8)]
 pub enum MimeType {
+    #[default]
     /// text/plain;charset=utf-8 mime type.
     ///
     /// The primary mime type used by most clients
-    TextPlainUtf8 = 0,
+    TextPlainUtf8,
     /// UTF8_STRING mime type.
     ///
     /// Some X11 clients are using only this mime type, so we
     /// should have it as a fallback just in case.
-    Utf8String = 1,
+    Utf8String,
     /// text/plain mime type.
     ///
     /// Fallback without charset parameter.
-    TextPlain = 2,
+    TextPlain,
+    /// Other mime type
+    Other(Cow<'static, str>),
+}
+
+impl AsRef<str> for MimeType {
+    fn as_ref(&self) -> &str {
+        match self {
+            MimeType::Other(s) => s.as_ref(),
+            m => ALLOWED_TEXT_MIME_TYPES[m.discriminant() as usize],
+        }
+    }
+}
+
+impl MimeType {
+    fn discriminant(&self) -> u8 {
+        unsafe { *(self as *const Self as *const u8) }
+    }
```

Comment:
No tricks like that please. You can do it safely e.g by slapping into `Text` enum and doing discriminant there.